docs/gke-deployment.md

```diff
@@ -0,0 +1,113 @@
+# Deploying k8-kate to Google Kubernetes Engine
+
+## Prerequisites
+
+- A GKE cluster (Standard or Autopilot).
+- `gcloud` CLI authenticated with `gcloud auth login`.
+- Local Docker environment (or Cloud Build) capable of building and pushing container images.
+- [`kubectl` configured](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl) to talk to your target cluster.
+- A Gemini API key with access to the model you plan to demo.
+
+## 1. Set your Google Cloud context
+
+```bash
+export PROJECT_ID="my-gcp-project"
+export REGION="us-central1"
+export CLUSTER_NAME="kubectl-ai-demo"
+
+gcloud config set project "${PROJECT_ID}"
+gcloud container clusters get-credentials "${CLUSTER_NAME}" --region "${REGION}"
+```
+
+These commands configure both `gcloud` and `kubectl` to operate on the cluster that will host `kubectl-ai`.
+
+## 2. Build and push the kubectl-ai image
+
+Pick an Artifact Registry repository or another container registry that your cluster can pull from. The snippet below creates an Artifact Registry repo (if needed), builds the image locally, and pushes it.
+
+```bash
+# Create an Artifact Registry (skip if you already have one)
+gcloud artifacts repositories create kubectl-ai \
+  --location="${REGION}" \
+  --repository-format=DOCKER \
+  --description="kubectl-ai demo images"
+
+# Configure Docker to authenticate to Artifact Registry
+gcloud auth configure-docker "${REGION}"-docker.pkg.dev
+
+# Build and push the container
+IMAGE="${REGION}-docker.pkg.dev/${PROJECT_ID}/kubectl-ai/kubectl-ai:latest"
+docker build -t "${IMAGE}" -f images/kubectl-ai/Dockerfile .
+docker push "${IMAGE}"
+```
+## 3. Prepare cluster namespaces and RBAC
+
+Create the namespaces and RBAC that the hosted agent requires:
+
+```bash
+# Namespace that hosts the kubectl-ai Deployment
+kubectl create namespace kubectl-ai
```

Comment:
should this namespace be part of `kubectl-ai-gke.yaml` to make it self-contained ?